test/testutils/testutils.cpp

```diff
@@ -106,5 +105,18 @@ namespace TestUtils {
             {"files", QVariantList{}}, {"dav", QVariantMap{{"chunking", "1.0"}}},
             {"checksums", QVariantMap{{"preferredUploadType", Utility::enumToString(algo)}, {"supportedTypes", algorithmNames}}}};
     }
+
+
+    AccountManagerRaii::AccountManagerRaii(const AccountStatePtr &accountStatePtr)
+        : _accountStatePrt(accountStatePtr)
+    {
+    }
+
+    AccountManagerRaii::~AccountManagerRaii()
+    {
+        if (_accountStatePrt) {
```

Comment:
Typo: `_accountStatePtr`.

Seems like a smart pointer would do the job just fine if you specify the deleter accordingly.